TheraLang.DAL/Repository/Repository.cs

```diff
@@ -9,46 +9,77 @@ namespace TheraLang.DAL.Repository
 {
     public class Repository<TEntity> : IRepository<TEntity> where TEntity : class
     {
+        private DbSet<TEntity> DbSet { get; }
         public Repository(DbSet<TEntity> dbSet)
         {
             DbSet = dbSet;
         }
+        public async Task<TEntity> GetAsync(Expression<Func<TEntity, bool>> where)
+        {
+            var entity = await DbSet.FirstOrDefaultAsync(where);
+            return entity;
+        }
+        public async Task<TEntity> GetAsync()
+        {
+            var entity = await DbSet.FirstOrDefaultAsync();
+            return entity;
+        }
 
-        private DbSet<TEntity> DbSet { get; }
+        public async Task<IQueryable<TEntity>> GetListAsync(Expression<Func<TEntity, bool>> where)
+        {
+            return DbSet.Where(where);
+        }
 
-        public IQueryable<TEntity> Get()
+        public async Task<IQueryable<TEntity>> GetListAsync()
         {
-            return DbSet;
+            return DbSet.AsQueryable();
         }
 
-        public void Remove(TEntity entity)
+        public async Task RemoveAsync(TEntity entity)
```

Comment:
```suggestion
        public Task RemoveAsync(TEntity entity)
        {
              DbSet.Remove(entity);
              return Task.CompletedTask;
        }
```

Reply:
just why?

Reply:
we can wrap up this into a task and await if you want to keep it